src/core/xconnection.rs

```diff
@@ -445,20 +482,32 @@ impl XcbConnection {
             root,
             check_win,
             atoms,
+            window_types,
             auto_float_types,
             randr_base,
         })
     }
 
-    fn atom(&self, name: &str) -> Result<u32> {
+    fn atom(&self, atom: &Atom) -> Result<u32> {
```

Comment:
If this function is now accepting a closed enum, it is no longer possible to fail (we will have errored on window manager initi when we tried to intern all of the atoms).